missing method GetHeader

go/pkg/mod/github.com/shiningrush/droplet/wrapper/gin@v0.3.0/gin.go:13:20: cannot use ctx.Writer (variable of type "github.com/gin-gonic/gin".ResponseWriter) as data.ResponseWriter value in struct literal: "github.com/gin-gonic/gin".ResponseWriter does not implement data.ResponseWriter (missing method GetHeader)

> hi, what version the gin used in your project is?

@ShiningRush I suffered such situaction as this issue metioned. I want to know. when I use
`

         github.com/shiningrush/droplet v0.4.2
	 github.com/shiningrush/droplet/wrapper/gin v0.3.0

`

error is:
`

        C:\go\pkg\mod\github.com\shiningrush\droplet\wrapper\gin@v0.3.0\gin.go:13:20: cannot use ctx.Writer (variable of type "github.com/gin-gonic/gin".ResponseWriter) as type data.ResponseWriter in struct literal:
	"github.com/gin-gonic/gin".ResponseWriter does not implement data.ResponseWriter (missing DelHeader method)


`

which gin version should I use.  because I need to use the droplet which has fixed #35 

as I use gin v1.8.1 as following. it appears this issue metioned problem.
![image](https://user-images.githubusercontent.com/48037235/232733748-4feae15d-a3f4-412b-be86-39d6d606191c.png)

In droplet(v0.4.2 which  solved issue 35). DelHeader is in data/resp.go , which is interface method.
![image](https://user-images.githubusercontent.com/48037235/232736491-a9ce4fae-8bad-4d1b-b6fb-7d75d975a852.png)


and droplet use  ctx.Write as RespWriter
![image](https://user-images.githubusercontent.com/48037235/232736887-00a4fa1a-5c5f-4425-8d25-85d8e8ee7b67.png)

I find in gin v1.8.1, ctx.Writer doesnot implment the DelHeader method.

--- a/wrapper/fasthttp/go.mod
+++ b/wrapper/fasthttp/go.mod
@@ -4,7 +4,7 @@ go 1.17
 
 require (
 	github.com/fasthttp/router v1.4.11
-	github.com/shiningrush/droplet v0.4.1
+	github.com/shiningrush/droplet v0.4.2
 	github.com/valyala/fasthttp v1.38.0
 )
 

--- a/wrapper/gin/go.mod
+++ b/wrapper/gin/go.mod
@@ -5,7 +5,7 @@ go 1.14
 require (
 	github.com/gin-gonic/gin v1.8.1
 	github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd // indirect
-	github.com/shiningrush/droplet v0.2.4
+	github.com/shiningrush/droplet v0.4.2
 )
 
-replace github.com/shiningrush/droplet => ../../
+//replace github.com/shiningrush/droplet => ../../

--- a/wrapper/gorestful/examples/main.go
+++ b/wrapper/gorestful/examples/main.go
@@ -1,0 +1,33 @@
+package main
+
+import (
+	"net/http"
+	"reflect"
+
+	"github.com/emicklei/go-restful/v3"
+	"github.com/shiningrush/droplet/core"
+	"github.com/shiningrush/droplet/wrapper"
+
+	rwrap "github.com/shiningrush/droplet/wrapper/gorestful"
+)
+
+func main() {
+	ws := new(restful.WebService)
+	ws.Route(ws.POST("/json_input/{id}").To(rwrap.Wraps(JsonInputDo, wrapper.InputType(reflect.TypeOf(&JsonInput{})))))
+	restful.Add(ws)
+	http.ListenAndServe(":8080", nil)
+}
+
+type JsonInput struct {
+	ID    string   `auto_read:"id,path" json:"id"`
+	User  string   `auto_read:"user,header" json:"user"`
+	IPs   []string `json:"ips"`
+	Count int      `json:"count"`
+	Body  []byte   `auto_read:"@body"`
+}
+
+func JsonInputDo(ctx core.Context) (interface{}, error) {
+	input := ctx.Input().(*JsonInput)
+
+	return input, nil
+}

--- a/wrapper/gorestful/go.mod
+++ b/wrapper/gorestful/go.mod
@@ -5,7 +5,7 @@ go 1.15
 require (
 	github.com/emicklei/go-restful/v3 v3.9.0
 	github.com/json-iterator/go v1.1.10 // indirect
-	github.com/shiningrush/droplet v0.2.4
+	github.com/shiningrush/droplet v0.4.2
 )
 
 replace github.com/shiningrush/droplet => ../../

